### PR TITLE
[WIP] Add first pass at db router [#OSF-7406]

### PR DESCRIPTION
**Pull request by cwisecarver**

### CODE REVIEW

File: osf/db/router.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        self._get_dsns()
+        global CACHED_MASTER
+        if not CACHED_MASTER:
+            CACHED_MASTER = self._get_master()
+
+    def _get_master(self):
+        for name, dsn in self.DSNS.iteritems():
+            conn = self._get_conn(dsn)
+            cur = conn.cursor()
+            cur.execute('SELECT pg_is_in_recovery();')
```
Comment (icereval) [inline osf/db/router.py]: This is good but lets stick with the JDBC implementation if at all possible.

https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java#L692

Comment (cwisecarver) [inline osf/db/router.py]: They're probably the same. My only concern would be a pedantic one. Yours checks if the server is read only. Mine checks to see if one is master vs one is slave. You can technically have a master that's read only. It's going to error either way though if you try to write to it.

**Comment by cwisecarver** [comment]
@icereval @chrisseto  Updated the query, added some comments, added setting to enable router, still running migrations so haven't actually tested running django with it.

**Comment by mfraezz** [comment]
Closing in favor of #6820

**Pull request closed without merging**